src/mlpack/methods/lars/lars.hpp

```diff
@@ -85,9 +82,14 @@ namespace mlpack {
  * }
  * @endcode
  */
+template<typename ModelMatType = arma::mat>
```

Comment:
The template parameter here doesn't refer to the type of the data but instead the type of the model; for LARS, depending on how you penalize (L1 or L2) you can get a dense or a sparse model.  So, the user can specify the type of storage they will use for the underlying model: `LARS<arma::mat>` or `LARS<arma::sp_mat>` (or `sp_fmat` or whatever).

But, this is different from the type of the data used for learning.  So, you can have a `LARS<arma::sp_mat>` that is L1-penalized, but then call `.Train()` with a dense `arma::mat`.  Or, vice versa.

This is the case with a bunch of model types, so I added `ModelMatType` as a parameter instead of just `MatType`, in those situations.

Reply:
Make sense to me, I got that eventually :+1: